### ISSUE TITLE
Extract more of the plumbing around entry points for reuse

### ISSUE THREAD

**Issue by imrekoszo** [lambdaisland/kaocha#312]
While working on [polylith-kaocha](https://github.com/imrekoszo/polylith-kaocha) I noticed that kaocha's 3 entry points (cli, -x, repl) seemingly reimplement common functionality with some subtle differences (like [cloverage not being usable from repl/run](https://github.com/lambdaisland/kaocha-cloverage/issues/16)). Because I wanted to support cloverage in p-k [I had to](https://clojurians.slack.com/archives/CCY2V0U6A/p1651252116723609?thread_ts=1651249678.157049&cid=CCY2V0U6A) again [reimplement](https://github.com/imrekoszo/polylith-kaocha/blob/a753ab14ab632bc913e782e5459296c6aff5659a/components/kaocha-wrapper/src/polylith_kaocha/kaocha_wrapper/runner.clj#L31-L67) some of the stuff that -x does.

It would be great from the perspective of other tooling calling kaocha in-process to have more of this common functionality available for reuse without having to be aware of internals. I imagine most of the benefits of this work would only be towards tooling authors mind you.

Apart from the wrappers around the main process with all hooks, one area that could be helpful is exposing the parsing of `-m`-style command line args so other command-line tooling calling kaocha [could pass them through](https://clojurians.slack.com/archives/CCY2V0U6A/p1663273604284869?thread_ts=1663220887.406909&cid=CCY2V0U6A) without calling -m or having to understand it.

Potentially related issue: #305

**Comment by imrekoszo** [comment]
I think it could be considered an indicator of success that there are more integrations than you had imagined. And the fact that all these implementations were able to function mean that the necessary puzzle pieces are available, which is great.

What I found missing are facades or pre-made compositions that let a caller

- expose pass-through cli configurability
- run stuff "as if it was run from -m" (plugins, hooks etc)
- but still be outside of kaocha being responsible for the entire process itself, like terminating it

**Comment by imrekoszo** [comment]
Related issue: https://github.com/imrekoszo/polylith-kaocha/issues/6

**Comment by alysbrooks** [comment]
I don't think this addresses the issue, but some of the refactoring work in #315 is in a similar spirit.
